### PR TITLE
PaymentSession: migrate getters to `java.time` API

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

**Review by msgilligan — CHANGES_REQUESTED** [review]
2 minor changes requested.

File: core/src/main/java/org/bitcoinj/protocols/payments/PaymentSession.java [hunk 1]
```diff
@@ -253,27 +255,44 @@ public Coin getValue() {
...
+     * Returns the expires time of the payment request, or empty if none.
      */
-    @Nullable public Date getExpires() {
+    public Optional<Instant> getExpiresInstant() {
         if (paymentDetails.hasExpires())
-            return new Date(paymentDetails.getExpires() * 1000);
+            return Optional.of(Instant.ofEpochSecond(paymentDetails.getExpires()));
         else
-            return null;
+            return Optional.empty();
```
[the remark this answers is not in the document]
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/protocols/payments/PaymentSession.java:277]: Yeah, that's again readabilty of the changeset vs end result.

File: core/src/main/java/org/bitcoinj/protocols/payments/PaymentSession.java [hunk 2]
```diff
@@ -253,27 +255,44 @@ public Coin getValue() {
...
+    }
+
+    /** @deprecated use {@link #getExpiresInstant()} */
+    @Nullable
+    @Deprecated
+    public Date getExpires() {
+        return getExpiresInstant()
+                .map(time -> Date.from(time))
```
[the remark this answers is not in the document]
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/protocols/payments/PaymentSession.java:285]: See error in the other PR.

[the remark this answers is not in the document]
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/protocols/payments/PaymentSession.java:285]: Ok thanks, in this case it worked.

**Comment by schildbach** [comment]
Rebased and merged.

**Pull request closed without merging**